### ISSUE TITLE
NVDA support for accessibility on Windows

### ISSUE THREAD

**Issue by StefanLobbenmeier** [StefanLobbenmeier/yt-dlp-compose#156]
Supposedly this is supported by jetpack compose, see https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-desktop-accessibility.html#enabling-accessibility-on-windows

I follow those steps, but it seems it is still disabled. Check with jetbrains why that is.

**Comment by StefanLobbenmeier** [comment]
@aryanchoudharypro

Can you share if there is a file in your user directory called `.accessibility.properties`? It should have been created by NVDA and contain something like this:

```
assistive_technologies=com.sun.java.accessibility.AccessBridge
screen_magnifier_present=true
```

See also https://youtrack.jetbrains.com/issue/CMP-373/Accessibility-doesnt-work-on-Windows-when-we-create-a-native-distribution#focus=Comments-27-11130024.0-0

**Comment by aryanchoudharypro** [comment]
yes, i do have the file with the same text

**Comment by aryanchoudharypro** [comment]
also when navigateing using alt + tab, the title of the window is reported as untitled

**Comment by aryanchoudharypro** [comment]
ok i have a question here. is the enabled or disabled state of the button  Embed Subtitles defined as accessibility info? asking this because nvda is able to read it. also we can test if the   bug is with jetpack compose or the code by adding a simple accessibility info, for example, download location, press enter to change

**Comment by StefanLobbenmeier** [comment]
Oh Thats Great News, I thought NVDA could not read anything and its a configuration hell, but if all I have to do is Go through Buttons and add Accessibility Information then I know what to do

Regarding your question, there is no explicit accessibility information in yet, but lots of components have some default semantics, and icons should have alt text already, but I can confirm it via https://developer.apple.com/documentation/accessibility/accessibility-inspector on my own machine

**Comment by aryanchoudharypro** [comment]
i can creat an issue about it lator, but can you tell me the directory structure i need to keep to make yt-dlp and ffmpeg work if put them manualy? i am asking this because its not able to download them. at startup, and i am not able to test the intire app

**Comment by StefanLobbenmeier** [comment]
Settings should be available from the start in the top right corner, in the settings there are dropdowns to confirm yt-dlp and ffmpeg from “DISK”, and then you can configure the path via an input or click the folder icon next the the input to select the folder in an explorer

**Comment by aryanchoudharypro** [comment]
ok so i am finding the settings window to be very inaccessable. for example, when i tab through a few options it says null, other times it doesn't say anything and sometimes reads system etc but its pritty hard to use as of now

**Comment by aryanchoudharypro** [comment]
when i finally clicked the disk option using ocr and stuff, i had an unlabled butten which i thaught to be the brows option. i clicked it and it exited with following error. SHCreateItemFromParsingName failed

**Comment by StefanLobbenmeier** [comment]
Hmm okay, I will check this with accessibility inspector

**Comment by StefanLobbenmeier** [comment]
Releasing a new version, its not perfect yet but now a lot of the ui elements have descriptions.

One thing I noticed is that with accessibility inspector, when a textfield in settings has a value, it no longer reads the label. I did not figure out how to prevent this yet. If it helps, the label is still available as the "Help" attribute. I could not figure out a way to fix that issue yet.

**Comment by aryanchoudharypro** [comment]
does seem a lot better now, how ever when i press tab to move to the next option, and i am on an edit feald, it gives the tab character to the edit feald instead of moveing the focus to the next option

**Comment by StefanLobbenmeier** [comment]
I see - the same happens for me in settings, but not in the main edit field for the url, so it should be possible to disable that behaviour
